conn.go

```diff
@@ -358,6 +359,15 @@ func (hc *halfConn) decrypt(b *block) (ok bool, prefixLen int, alertValue alert)
 				hc.additionalData[11] = byte(n >> 8)
 				hc.additionalData[12] = byte(n)
 				additionalData = hc.additionalData[:]
+			} else {
+				if len(payload) > int(uint16(1<<14)+256) {
```

Comment:
These typecasts are confusing. Everything is already an `int`. Maybe just  `if len(payload) > ((1 << 14) + 256) {`.